packages/scandipwa/src/route/Checkout/Checkout.container.js

```diff
@@ -251,13 +254,55 @@ export class CheckoutContainer extends PureComponent {
     }
 
     componentDidUpdate(prevProps, prevState) {
-        const { match: { params: { step: urlStep } }, isEmailAvailable, updateEmail } = this.props;
-        const { match: { params: { step: prevUrlStep } } } = prevProps;
-        const { email } = this.state;
+        const {
+            match: {
+                params: {
+                    step: urlStep = ''
+                }
+            },
+            isEmailAvailable,
+            updateEmail,
+            isCartLoading,
+            shippingFields,
+            shippingFields: {
+                city
+            },
+            totals: {
+                is_virtual
+            },
+            showInfoNotification,
+            history
+        } = this.props;
+
+        const {
+            match: {
+                params: {
+                    step: prevUrlStep = ''
+                }
+            },
+            isCartLoading: prevIsCartLoading
+        } = prevProps;
+
+        const { email, checkoutStep } = this.state;
         const { email: prevEmail } = prevState;
 
+        this.handleRedirectIfNoItemsInCart();
+
+        if (prevIsCartLoading && !isCartLoading) {
+            if (!city && !is_virtual && checkoutStep === BILLING_STEP) {
```

Comment:
Wouldn't it break if the step is `SUCCESS_STEP`?

Reply:
No, because `this.handleRedirectIfNoItemsInCart();` will work first and will redirect to cart on reloading `/success`

-- Marked the lines with a 'Here' comment :)